src/blob.rs

```diff
@@ -31,6 +31,12 @@ use crate::log::LogExt;
 pub struct BlobObject<'a> {
     blobdir: &'a Path,
     name: String,
+
+    /// On some platforms images are sent as
+    /// [`crate::message::Viewtype::Sticker`]. [`Self::recode_to_size()`] rechecks if it is a true
+    /// sticker assuming that it must have at least one fully transparent corner, otherwise this
+    /// flag is reset.
+    pub maybe_sticker: bool,
```

Comment:
Decided to pass it to `recode_to_image_size()` as `&mut bool`, this way it looks to me more readable -- there is no additional return value. And if it was returned as a tuple member, it would be unnamed.